test/Microsoft.Extensions.Primitives.Tests/ChangeTokenTest.cs

```diff
@@ -87,5 +88,37 @@ public void ChangesFireAfterExceptionsWithState()
             Assert.Equal(2, count);
             Assert.NotNull(callbackState);
         }
+
+        [Fact]
+        public void AsyncLocalsNotCapturedAndRestored()
+        {
+            // Capture clean context
+            var executionContext = ExecutionContext.Capture();
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            var cancellationToken = cancellationTokenSource.Token;
+            var cancellationChangeToken = new CancellationChangeToken(cancellationToken);
+            
+            // Set AsyncLocal
+            var asyncLocal = new AsyncLocal<int>();
+            asyncLocal.Value = 1;
+
+            // Register Callback
+            cancellationChangeToken.RegisterChangeCallback(al =>
+            {
+                // AsyncLocal not set, when run on clean context
+                // A suppressed flow runs in current context, rather than restoring the captured context
+                Assert.Equal(0, ((AsyncLocal<int>) al).Value);
```

Comment:
Might want to set something that verifies that this block of code was executed.

```C#
var executed = false;
...
     Assert.Equal(0, ((AsyncLocal<int>) al).Value);
     executed = true;
...

Assert.True(executed);
```

Reply:
Oh, I like that :)